src/database/database.cpp

```diff
@@ -0,0 +1,1149 @@
+}
+
+Error Database::GetAllInstances(Array<sm::launcher::InstanceData>& instances)
+{
+    LOG_DBG() << "Get all instances";
+
+    try {
+        std::vector<DBInstanceData::Fields> result;
+
+        *mSession << "SELECT * FROM instances;", into(result), now;
+
+        for (const auto& info : result) {
+            if (auto err = instances.PushBack(DBInstanceData::ToAos(info)); !err.IsNone()) {
+                return AOS_ERROR_WRAP(Error(err, "db instances count exceeds application limit"));
+            }
+        }
+    } catch (const Poco::Exception& e) {
+        return AOS_ERROR_WRAP(Error(ErrorEnum::eFailed, e.what()));
+    }
+
+    return ErrorEnum::eNone;
+}
+
+RetWithError<uint64_t> Database::GetOperationVersion() const
+{
+    LOG_DBG() << "Get operation version";
+
+    RetWithError<uint64_t> result {0, ErrorEnum::eNone};
+
+    try {
+        Poco::Data::Statement statement {*mSession};
+
+        statement << "SELECT operationVersion FROM config;", into(result.mValue);
+
+        if (statement.execute() == 0) {
+            return {0, ErrorEnum::eNotFound};
+        }
+    } catch (const Poco::Exception& e) {
+        result.mError = AOS_ERROR_WRAP(Error(ErrorEnum::eFailed, e.what()));
+    }
+
+    return result;
+};
+
+Error Database::SetOperationVersion(uint64_t version)
+{
+    LOG_DBG() << "Set operation version: version=" << version;
+
+    try {
+        *mSession << "UPDATE config SET operationVersion = ?;", use(version), now;
+    } catch (const Poco::Exception& e) {
+        return AOS_ERROR_WRAP(Error(ErrorEnum::eFailed, e.what()));
+    }
+
+    return ErrorEnum::eNone;
+}
+
+Error Database::GetOverrideEnvVars(cloudprotocol::EnvVarsInstanceInfoArray& envVarsInstanceInfos) const
+{
+    LOG_DBG() << "Get override env vars";
+
+    try {
+        std::string envVarsJson;
+
+        *mSession << "SELECT envvars FROM config;", into(envVarsJson), now;
+
+        return ConvertEnvVarsInstanceInfoArrayFromJSON(envVarsJson, envVarsInstanceInfos);
+    } catch (const Poco::Exception& e) {
+        return AOS_ERROR_WRAP(Error(ErrorEnum::eFailed, e.what()));
+    }
+
+    return ErrorEnum::eNone;
+}
+
+Error Database::SetOverrideEnvVars(const cloudprotocol::EnvVarsInstanceInfoArray& envVarsInstanceInfos)
+{
+    LOG_DBG() << "Set override env vars";
+
+    try {
+        const auto envVarInstanceInfosJson
+            = common::utils::Stringify(ConvertEnvVarsInstanceInfoArrayToJSON(envVarsInstanceInfos));
+
+        Poco::Data::Statement statement {*mSession};
+
+        statement << "UPDATE config SET envvars = ?;", bind(envVarInstanceInfosJson);
+
+        if (statement.execute() == 0) {
+            return AOS_ERROR_WRAP(ErrorEnum::eNotFound);
+        }
+    } catch (const Poco::Exception& e) {
+        return AOS_ERROR_WRAP(Error(ErrorEnum::eFailed, e.what()));
+    }
+
+    return ErrorEnum::eNone;
+}
+
+RetWithError<Time> Database::GetOnlineTime() const
+{
+    LOG_DBG() << "Get online time";
```

Comment:
Also would be nince to see the time for debug purposes. Move after read from db.